src/library/trackset/setlogfeature.h

```diff
@@ -20,16 +20,17 @@ class SetlogFeature : public BasePlaylistFeature {
 
     void bindLibraryWidget(WLibrary* libraryWidget,
             KeyboardEventFilter* keyboard) override;
-    void bindSidebarWidget(WLibrarySidebar* pSidebarWidget) override;
+    void activatePlaylist(int playlistId) override;
 
   public slots:
     void onRightClick(const QPoint& globalPos) override;
     void onRightClickChild(const QPoint& globalPos, QModelIndex index) override;
     void slotJoinWithPrevious();
     void slotGetNewPlaylist();
+    void activate() override;
 
   protected:
-    QList<BasePlaylistFeature::IdAndLabel> createPlaylistLabels() override;
+    QModelIndex constructChildModel(int selectedId);
```

Comment:
Missing `override`

Reply:
```
In file included from /home/poelzi/Projects/mixxx/src/library/library.cpp:32:
/home/poelzi/Projects/mixxx/src/library/trackset/setlogfeature.h:33:17: error: ‘QModelIndex SetlogFeature::constructChildModel(int)’ marked ‘override’, but does not override
     QModelIndex constructChildModel(int selectedId) override;
                 ^~~~~~~~~~~~~~~~~~~

```
Does not work. constructChildModel is unique to PlaylistFeature and SetlogFeature, but this function does not exist in BasePlaylistFeature.

Reply:
What was the reason for removing it from BasePlaylistModel instead of declaring it as pure virtual? Non-virtual, independent methods of the same name in derived classes are totally valid, but could be confusing.

Reply:
Otherwise I would had to implement createPlaylistLabels() which the setlogfeature does not need. Thats why the two distinct implementations when into their respective child classes.